Fixed #31026 -- Switch form rendering to template engine (#12133)
django/django: docs/ref/forms/renderers.txt

Using this renderer requires you to make sure the form templates your project
needs can be located.

Context available in form templates
===================================

.. versionadded:: 4.0

Form templates receive a context from :meth:`.Form.get_context`. Be default,

> forms receive dictionary with the following values:

Reviewer: a dictionary

> forms receive a dictionary with the following values:

Reviewer 2: ```suggestion
this context is a dictionary with the following entries:
```

that subclass ``Input`` defines ``widget['type']`` and :class:`.MultiWidget`
defines ``widget['subwidgets']`` for looping purposes.

Overriding built-in form templates
==================================

.. versionadded:: 4.0

.. _overriding-built-in-form-templates:

:attr:`.Form.template_name`

To override form templates, you must use the :class:`TemplatesSetting`

> renderer. Then overriding widget templates works :doc:`the same as

Reviewer 2: This is unclear; I'd expect to see a listing of the relevant template names here
